internal/server.go

```diff
@@ -0,0 +1,44 @@
+package internal
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/rs/zerolog"
+)
+
+func StartServer(logger zerolog.Logger, filePath string, port int) error {
```

Comment:
Let's separate it into two functions, NewServer (on the package) and StartServer on the server's struct

Also, let's make sure the options are abstracted out into another struct e.g ServerCfgOptions maybe